### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by bryanhce**
Update target user profile, value proposition, use cases, non-functional requirements and glossary.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -283,16 +312,53 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    Use case ends.
+
+* 3a. The inputted fields are invalid
 
-**Use case: Delete a person**
+  * 3a1. GuestList shows an error message.
+  
+    Use case resumes at step 2.
+  
+**Use case: Delete a guest**
```
Comment (jehousoh) [inline docs/DeveloperGuide.md:339]: Should we shift the use cases to be ordered based on what function comes first like our UG? ie Add -> Delete -> List -> Exit

**Review by jehousoh — CHANGES_REQUESTED** [review]
LGTM other than the order of the use cases!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (bryanhce) [reply docs/DeveloperGuide.md:339]: Yes, we should follow that order. Will change.

**Review by jehousoh — APPROVED** [review]
lgtm!

**Pull request merged**